lint/pyproject.toml

```diff
@@ -0,0 +1,12 @@
+# This file is part of sv-witnesses repository: https://github.com/sosy-lab/sv-witnesses
+#
+# SPDX-FileCopyrightText: 2021 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+[build-system]
+requires = [
+    'setuptools >= 42.0.0',
+    'wheel >= 0.32.0',
+]
+build-backend = 'setuptools.build_meta'
```

Comment:
This is basically the old and complicated way of building Python packages. For modern tools without backwards-compatibility requirements, there are several newer approaches that aim at drastically simplifying this, like [Poetry](https://python-poetry.org/) but IIRC also others. I do not have experience with these tools, but I guess we should investigate them and choose one to keep future maintenance less effort.

Reply:
@lembergerth Maybe you have an overview over good Python packaging tools, or experience with them?

Reply:
I've researched this when I wanted to deploy pybib2web.
Poetry was the best alternative to setuptools, but I decided against it.
If I recall correctly, Poetry basically requires all metadata and configuration to be duplicated in separate poetry-blocks
(for example [here](https://python-poetry.org/docs/pyproject/#classifiers), the classifiers for pypi must be repeated in a `[tool.poetry]` block).

Because of that, I decided to go with a single setup.cfg that can also hold the configuration options for pylint, pytest, etc,
and switch to pyproject.toml once it can fully replace it.
So I stayed with setuptools.

Reply:
Since it has been some time I looked into Poetry again, and there does not seem any duplication to be necessary. It is true that the metadata has to be given in `[tool.poetry]` blocks in the `pyproject.toml`, but at least for this project we no longer need the `setup.cfg` afterwards.